Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/PhysicObject.cpp

```diff
@@ -0,0 +1,152 @@
+#include "PhysicObject.hpp"
+#include "../architecture/util.hpp"
+#include "SphereCollider.hpp"
+#include "BoxCollider.hpp"
+#include "SAT.hpp"
+
+namespace sne
+{
+
+    namespace saintNecPhysics
+    {
+        PhysicObject::PhysicObject(float mass) : PhysicObject({0, 0, 0}, mass)
+        {
+        }
+
+        PhysicObject::PhysicObject(const glm::vec3 &position, float mass) : _position(position), _mass(mass)
+        {
+            if (mass <= 0)
+                throw IllegalMassException();
+        }
+
+        const glm::vec3 &PhysicObject::getAcceleration() const
+        {
+            return _acceleration;
+        }
+
+        const glm::vec3 &PhysicObject::getVelocity() const
+        {
+            return _velocity;
+        }
+
+        const glm::vec3 &PhysicObject::getPosition() const
+        {
+            return _position;
+        }
+
+        const glm::vec3 &PhysicObject::getRotation() const
+        {
+            return _rotation;
+        }
+
+        float PhysicObject::getAmortissement() const
+        {
+            return _amortissement;
+        }
+
+        float PhysicObject::getMass() const
+        {
+            return _mass;
+        }
+
+        void PhysicObject::setAcceleration(const glm::vec3 &a)
+        {
+            _acceleration = a;
+        }
+
+        void PhysicObject::setVelocity(const glm::vec3 &v)
+        {
+            _velocity = v;
+        }
+
+        void PhysicObject::setPosition(const glm::vec3 &p)
+        {
+            _position = p;
+        }
+
+        void PhysicObject::setRotation(const glm::vec3 &r)
+        {
+            _rotation = r;
+        }
+
+        void PhysicObject::setMass(float m)
+        {
+            _mass = m;
+        }
+
+        void PhysicObject::setCollider(Collider *collider)
+        {
+            _collider = collider;
+        }
+
+        void PhysicObject::applyForce(const Force &f)
+        {
+            _acceleration += f / _mass;
+        }
+
+        void PhysicObject::compute(float dt)
+        {
+            _position += _velocity * dt + ((float)0.5) * _acceleration * dt * dt;
+            _velocity += _acceleration * dt;
+
+            if (_collider)
+                _collider->setCenter(_position);
+        }
+
+        void PhysicObject::update()
+        {
+            compute(1 / FPS);
+        }
+
+        void PhysicObject::computeCollide(PhysicObject &obj)
+        {
+            if (!_collider || !obj._collider)
+                throw std::exception_ptr();
+
+            try
+            {
+                if (!_collider->collide(obj._collider))
+                    return;
+            }
+            catch (const SATIllegalUseException &e)
+            {
+                std::cout << e.what() << '\n';
```

Comment:
IS it possible to add details for the exception -> get the gameobject name for example